griptape/tasks/base_control_flow_task.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import annotations
+from abc import ABC
+from attrs import define
+from griptape.tasks import BaseTask
+from griptape.memory.meta import ControlFlowMetaEntry
+
+
+@define
+class BaseControlFlowTask(BaseTask, ABC):
+    def before_run(self) -> None:
+        super().before_run()
+
+        self.structure.logger.info(f"{self.__class__.__name__} {self.id}\nInput: {self.input.to_text()}")
+
+    def after_run(self) -> None:
+        super().after_run()
+
+        self.structure.meta_memory.add_entry(
+            ControlFlowMetaEntry(
+                input_tasks=[parent.id for parent in self.parents],
+                output_tasks=[child.id for child in filter(lambda child: not child.is_finished(), self.children)],
+                output=self.output,
+            )
+        )
```

Comment:
Why are we adding to Meta Memory at all? What does the LLM need to know about the branch decisions?
Also, there has been talk of removing Meta Memory entirely, I'm not sure we should rely on it.

Reply:
wanted to have some mechanism of storing the "decision" of the control flow.